pandas/core/arrays/arrow/array.py

```diff
@@ -2088,6 +2091,62 @@ def _dt_isocalendar(self):
     def _dt_is_leap_year(self):
         return type(self)(pc.is_leap_year(self._pa_array))
 
+    @property
+    def _dt_is_month_start(self):
+        return type(self)(pc.equal(pc.day(self._pa_array), 1))
+
+    @property
+    def _dt_is_month_end(self):
+        plus_one_day = pc.add(self._pa_array, pa.scalar(datetime.timedelta(days=1)))
+        return type(self)(pc.equal(pc.day(plus_one_day), 1))
+
+    @property
+    def _dt_is_year_start(self):
+        return type(self)(
+            pc.and_(
+                pc.equal(pc.month(self._pa_array), 1),
+                pc.equal(pc.day(self._pa_array), 1),
+            )
+        )
+
+    @property
+    def _dt_is_year_end(self):
+        return type(self)(
+            pc.and_(
+                pc.equal(pc.month(self._pa_array), 12),
+                pc.equal(pc.day(self._pa_array), 31),
+            )
+        )
+
+    @property
+    def _dt_is_quarter_start(self):
+        is_correct_month = pc.is_in(pc.month(self._pa_array), pa.array([1, 4, 7, 10]))
+        is_first_day = pc.equal(pc.day(self._pa_array), 1)
+        return type(self)(pc.and_(is_correct_month, is_first_day))
+
+    @property
+    def _dt_is_quarter_end(self):
+        is_correct_month = pc.is_in(pc.month(self._pa_array), pa.array([3, 6, 9, 12]))
+        plus_one_day = pc.add(self._pa_array, pa.scalar(datetime.timedelta(days=1)))
+        is_first_day = pc.equal(pc.day(plus_one_day), 1)
+        return type(self)(pc.and_(is_correct_month, is_first_day))
+
+    @property
+    def _dt_days_in_month(self):
+        pa_array = self._pa_array
+        if self.dtype.pyarrow_dtype.unit != "ns":
+            pa_array = self._pa_array.cast(
+                pa.timestamp("ns", tz=self.dtype.pyarrow_dtype.tz)
+            )
+        pa_array_int = pa_array.cast(pa.int64())
+        if self._hasna:
+            pa_array_int = pa_array_int.fill_null(iNaT)
+        np_result = get_date_field(pa_array_int.to_numpy(), "dim")
+        mask = np_result == -1
+        return type(self)(pa.array(np_result, mask=mask))
```

Comment:
This might make sense to implement upstream. Otherwise this might work:
```python
pc.days_between(pc.floor_temporal(self._pa_array, unit='month'), pc.ceil_temporal(self._pa_array, unit='month'))
```